File: algorithms/easy/maximum_odd_binary_number.dart
class Solution {
  String maximumOddBinaryNumber(String s) {
    String l = '';
    String r = '';
    for (int i = 0; i < s.length; i++) {
      final c = s[i];
      if (c == '1') {
        l += c;
      } else {
        r += c;
      }
    }
    if (l.length == 1) {
      return r + l;
    }
    return l.substring(0, l.length - 1) + r + l.substring(l.length - 1);
  }
}

File: algorithms/easy/maximum_odd_binary_number.rs
impl Solution {
    pub fn maximum_odd_binary_number(s: String) -> String {
        let mut l = String::new();
        let mut r = String::new();
        for c in s.chars() {
            if c == '1' {
                l.push(c);
            } else {
                r.push(c);
            }
        }
        if l.len() == 1 {
            format!("{}{}", r, l)
        } else {
            format!("{}{}{}", &l[..l.len() - 1], r, &l[l.len() - 1..])
        }
    }
}
